src/components/FlowbiteIllustration.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+export const FlowbiteIllustration = ({
+	name,
+	className = '', 
+	width = '300px',
+	height = 'auto',
+	title = '',
+	basePath = '/media/illustrations',
+}) => {
+	// Sanitize the basePath to prevent path traversal
+	const sanitizedBasePath = basePath
+		?.replace(/\.\./g, '') // Remove path traversal attempts
+		?.replace(/\/+/g, '/') // Replace multiple slashes with single slash
+		?.replace(/\/$/, ''); // Remove trailing slash
+
+	// Sanitize the name to prevent path traversal and ensure valid filename
+	const sanitizedName = name
+		?.replace(/[^a-zA-Z0-9\-_]/g, '') // Only allow alphanumeric, hyphens, underscores
+		?.toLowerCase(); // Normalize to lowercase
+
+	if (!sanitizedBasePath) {
```

Comment:
You should move this just below `sanitizedBasePath` that you dont proceed with name sanitization if the base path will be corrupted